src/interfaces/assistants_web/src/utils/cohereColors.ts

```diff
@@ -0,0 +1,286 @@
+  'border-evolved-mushroom-500 dark:border-evolved-mushroom-500',
+  'border-evolved-green-500 dark:border-evolved-green-500',
+  'border-mushroom-700 dark:border-mushroom-700',
+];
+
+// DEFAULT COLORS
+const DEFAULT_BG_COLOR = 'bg-evolved-blue-500 dark:bg-evolved-blue-500';
+const DEFAULT_TEXT_COLOR = 'text-evolved-blue-500 dark:text-blue-700';
+const DEFAULT_FILL_COLOR = 'fill-evolved-blue-500 dark:fill-blue-700';
+const DEFAULT_CONTRAST_TEXT_COLOR = 'text-blue-800 dark:text-blue-800';
+const DEFAULT_BORDER_COLOR = 'border-evolved-blue-500 dark:border-blue-500';
+const HOVER_DEFAULT_BG_COLOR = 'hover:bg-evolved-blue-500 dark:hover:bg-evolved-blue-500';
+const HOVER_DEFAULT_TEXT_COLOR = 'hover:text-evolved-blue-500 dark:hover:text-blue-700';
+const HOVER_DEFAULT_FILL_COLOR = 'hover:fill-evolved-blue-500 dark:hover:fill-blue-700';
+const HOVER_DEFAULT_CONTRAST_TEXT_COLOR = 'hover:text-blue-800 dark:hover:text-blue-800';
+const HOVER_DEFAULT_BORDER_COLOR = 'hover:border-evolved-blue-500 dark:hover:border-blue-500';
+
+const COLORS_MAP = {
+  text: {
+    default: DEFAULT_TEXT_COLOR,
+    defaultHover: HOVER_DEFAULT_TEXT_COLOR,
+    brandedHover: HOVER_TEXT_COLOR_LIST,
+    branded: TEXT_COLOR_LIST,
+  },
+  fill: {
+    default: DEFAULT_FILL_COLOR,
+    defaultHover: HOVER_DEFAULT_FILL_COLOR,
+    brandedHover: HOVER_FILL_COLOR_LIST,
+    branded: FILL_COLOR_LIST,
+  },
+  border: {
+    default: DEFAULT_BORDER_COLOR,
+    defaultHover: HOVER_DEFAULT_BORDER_COLOR,
+    brandedHover: HOVER_BORDER_COLOR_LIST,
+    branded: BORDER_COLOR_LIST,
+  },
+  background: {
+    default: DEFAULT_BG_COLOR,
+    defaultHover: HOVER_DEFAULT_BG_COLOR,
+    brandedHover: HOVER_BG_COLOR_LIST,
+    branded: BG_COLOR_LIST,
+  },
+  contrastText: {
+    default: DEFAULT_CONTRAST_TEXT_COLOR,
+    defaultHover: HOVER_DEFAULT_CONTRAST_TEXT_COLOR,
+    brandedHover: HOVER_CONTRAST_TEXT_COLOR_LIST,
+    branded: CONTRAST_TEXT_COLOR_LIST,
+  },
+};
+
+/**
+ * @description Get a color from the Cohere color palette, when no index is provided, a random color is returned
+ * @param id - id for generating a constant color in the palette
+ * @param options - options for the color, text, border, or background
+ * @param options.text - if true, returns a text color
+ * @param options.fill - if true, returns a fill color
+ * @param options.border - if true, returns a border color
+ * @param options.background - if true, returns a background color
+ * @param options.contrastText - if true, returns a contrast text color
+ * @param variant.hover - if true, returns a hover color
+ * @param variant.theme - if true, returns the theme name
+ * @returns color from the Cohere color palette
+ */
+export const getCohereColor = (
+  id: string | undefined,
+  options: {
+    text?: boolean;
+    fill?: boolean;
+    border?: boolean;
+    background?: boolean;
+    contrastText?: boolean;
+  },
+  variant?: { hover?: boolean }
+): string => {
+  if (id === undefined) {
```

Comment:
so many deep ifs.. 😱

Reply:
yes, I have an idea to refactor all of this. On follow up PR, let's see